src/main/kotlin/org/opensearch/replication/task/shard/ShardReplicationTask.kt

```diff
@@ -196,9 +196,14 @@ class ShardReplicationTask(id: Long, type: String, action: String, description:
         updateTaskState(FollowingState)
         val followerIndexService = indicesService.indexServiceSafe(followerShardId.index)
         val indexShard = followerIndexService.getShard(followerShardId.id)
-        // Renewing retention lease with retain all. After the task starts getting changes, retention lease would be
-        // renewed based on lastSyncedGlobalCheckpoint.
-        retentionLeaseHelper.renewRetentionLease(leaderShardId, RetentionLeaseActions.RETAIN_ALL , followerShardId)
+        // Adding retention lease at lastSyncedGlobalCheckpoint. This makes sure
+        // new tasks spawned after node changes/shard movements are handled properly
+        try {
+            logInfo("Adding retention lease")
+            retentionLeaseHelper.addRetentionLease(leaderShardId, indexShard.lastSyncedGlobalCheckpoint, followerShardId)
```

Comment:
should we just renew here ?

Reply:
As discussed offline, we need to add lease here because lease gets cancelled with task cancellation. So when task spins up on a new node, it needs to again add the retention lease.